pkg/publish/ingress.go

```diff
@@ -21,27 +24,58 @@ import (
 	kclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func toPrefix(domain, serviceName string, appInstance *v1.AppInstance) (hostPrefix string) {
-	if strings.HasSuffix(domain, "on-acorn.io") {
-		var appInstanceIDSegment string
-		var appInstanceIDSegmentByte [32]byte
+var ErrPatternParseFailed = errors.New("failed to parse endpoint pattern")
 
-		appInstanceIDSegment = serviceName + ":" + appInstance.GetName() + ":" + appInstance.Namespace
-		appInstanceIDSegmentByte = sha256.Sum256([]byte(appInstanceIDSegment))
-		appInstanceIDSegment = hex.EncodeToString(appInstanceIDSegmentByte[:])[:12]
-		hostPrefix = name.Limit(serviceName+"-"+appInstance.GetName(), 62-len(appInstanceIDSegment)) + "-" + appInstanceIDSegment
+func toEndpoint(domain, serviceName string, appInstance *v1.AppInstance) (endpoint string, err error) {
```

Comment:
Some changes:
1. the pattern only comes from the install config. dont want one on the app
2. I just want there to ALWAYS be a pattern, defaulting to the Let's Encrypt version. This is the current on-acorn.io logic: con-app-hash.cluster-domain. ...Note that cluster-domain in those scenarios includes the domain assigned from acorn-dns, so its actually like `.kaj198xcvxcv.on-acorn.io`
3. That bit of logic that does `62-len(...)` is because the max length of a segment of a domain is 63 characters (I don't know why it isnt `63-len()`). But in the pattern based logic, we are losing our protection against that. It would be good to not lose that. But this gets fairly tricky if we're using a pattern. We could, perhaps, just build the domain and check each segment after its built and return an error if its too long.